File: Go/Network/get-ip-type.go
package main

import (
	"net"
	"fmt"
	"strings"
)

func main() {
	// Check what type of IP address is.
	fmt.Println(checkIPType(net.ParseIP("1.1.1.1")))
	fmt.Println(checkIPType(net.ParseIP("2606:4700:4700::1111")))
}

// Check whether the given IP address is either IPv4 or IPv6.
func checkIPType(userIP net.IP) string {
	if strings.Contains(string(userIP), ".") {
		return "IPv4"
	} else if strings.Contains(string(userIP), ":") {
		return "IPv6"
	}
	return "Unknown"
}

File: TypeScript/Network/get-ip-type.ts
function main(): void {
    // Get the type of a given IP.
    console.log(getIPType("1.1.1.1"))
}

main()

// Get the type of a given ip, IPV4; IPV6
function getIPType(content: string): string {
    if (content.includes(".")) {
        return "ipv4"
    } else if (content.includes(":")) {
        return "ipv4"
    }
    return "unknown"
}
